cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/IndexCreator.java

```diff
@@ -143,6 +158,7 @@ private String ensureIndexed(final Index index) {
             return null;
         }
 
+        final Index indexToCreate = index;
```

Comment:
I'd be tempted to instead change the argument to `proposedIndex` and this variable to `index`, I think that's clearer.